manticore/models.py

```diff
@@ -0,0 +1,80 @@
+from .utils.helpers import issymbolic
+from .core.smtlib.solver import solver
+from .core.smtlib.operators import ITEBV, ZEXTEND
+
+
+def _find_zero(cpu, constrs, ptr):
+    """
+    Helper for finding the closest NULL or, effectively NULL byte from a starting address.
+
+    :param Cpu cpu:
+    :param ConstraintSet constrs: Constraints for current `State`
+    :param int ptr: Address to start searching for a zero from
+    :return: Offset from `ptr` to first byte that is 0 or an `Expression` that must be zero
+    """
+
+    offset = 0
+    while True:
+        byt = cpu.read_int(ptr + offset, 8)
+
+        if issymbolic(byt):
+            if not solver.can_be_true(constrs, byt != 0):
```

Comment:
Must be zero. if we did might be zero, the tree strcmp would build would be incomplete if that byte was not zero (and the string kept going) (since strcmp starts at the zero and works its way back)